clar2wasm/tests/wasm-generation/principal.rs

```diff
@@ -0,0 +1,230 @@
+//
+// Proptests that should only be executed
+// when running Clarity::V2 or Clarity::v3.
+//
+#[cfg(not(feature = "test-clarity-v1"))]
+mod clarity_v2_v3 {
+    use clar2wasm::tools::{crosscheck, crosscheck_with_network, Network};
+    use clarity::util::secp256k1::{Secp256k1PrivateKey, Secp256k1PublicKey};
+    use clarity::vm::types::{
+        ASCIIData, BuffData, CharType, OptionalData, PrincipalData, QualifiedContractIdentifier,
+        SequenceData, StandardPrincipalData, TupleData,
+    };
+    use clarity::vm::Value;
+    use proptest::prelude::{any, Just, Strategy};
+    use proptest::string::string_regex;
+    use proptest::{option, prop_oneof, proptest};
+
+    use crate::{buffer, runtime_config, standard_principal, PropValue};
+
+    fn strategies_for_version_byte() -> impl Strategy<Value = i32> {
+        prop_oneof![
+            25 => Just(0x1A),
+            25 => Just(0x15),
+            50 => 0x00..=0x1F
+        ]
+    }
+
+    fn create_principal(version: u8, principal: &Vec<u8>, contract_name: Option<&str>) -> Value {
+        if let Some(contract_name) = contract_name {
+            // contract principal requested
+            Value::Principal(PrincipalData::Contract(QualifiedContractIdentifier::new(
+                StandardPrincipalData(
+                    version,
+                    principal
+                        .as_slice()
+                        .try_into()
+                        .expect("slice bigger than 20 bytes"),
+                ),
+                contract_name.into(),
+            )))
+        } else {
+            // standard principal requested
+            Value::Principal(PrincipalData::Standard(StandardPrincipalData(
+                version,
+                principal
+                    .as_slice()
+                    .try_into()
+                    .expect("slice bigger than 20 bytes"),
```

Comment:
Code duplication for the `StandardPrincipalData`

Reply:
Fixed.